pkg/controller/jobframework/jobframework_test.go

```diff
@@ -0,0 +1,175 @@
+						mapper.Add(gvk, apimeta.RESTScopeNamespace)
+					}
+					return mapper, nil
+				},
+			}
+			mgr, err := ctrlmgr.New(&rest.Config{}, mgrOpts)
+			if err != nil {
+				t.Fatalf("Failed to setup manager: %v", err)
+			}
+
+			certsReady := make(chan struct{})
+			close(certsReady)
+			gotError := SetupControllers(mgr, logger, tc.modifyOpts, certsReady, tc.opts...)
+			if diff := cmp.Diff(tc.wantError, gotError, cmpopts.EquateErrors()); len(diff) != 0 {
+				t.Errorf("Unexpected error from SetupControllers (-want,+got):\n%s", diff)
+			}
+		})
+	}
+}
+
+func TestSetupIndexes(t *testing.T) {
+	testNamespace := "test"
+
+	cases := map[string]struct {
+		opts                  []Option
+		wls                   []client.Object
+		filter                client.ListOption
+		wantError             error
+		wantFieldMatcherError bool
+		wantList              []string
+	}{
+		"proper indexes are set": {
+			wls: []client.Object{
+				utiltesting.MakeWorkload("alpha-wl", testNamespace).
+					OwnerReference(batchv1.SchemeGroupVersion.WithKind("Job"), "alpha", "job", true, true).
+					Obj(),
+				utiltesting.MakeWorkload("beta-wl", testNamespace).
+					OwnerReference(batchv1.SchemeGroupVersion.WithKind("Job"), "beta", "job", true, true).
+					Obj(),
+			},
+			opts: []Option{
+				WithEnabledFrameworks(sets.New("batch/job")),
+			},
+			filter:   client.MatchingFields{GetOwnerKey(batchv1.SchemeGroupVersion.WithKind("Job")): "alpha"},
+			wantList: []string{"alpha-wl"},
+		},
+		"kubeflow.org/mpijob is disabled in the configAPI": {
+			wls: []client.Object{
+				utiltesting.MakeWorkload("alpha-wl", testNamespace).
+					OwnerReference(kubeflow.SchemeGroupVersionKind, "alpha", "mpijob", true, true).
+					Obj(),
+				utiltesting.MakeWorkload("beta-wl", testNamespace).
+					OwnerReference(kubeflow.SchemeGroupVersionKind, "beta", "mpijob", true, true).
+					Obj(),
+			},
+			opts: []Option{
+				WithEnabledFrameworks(sets.New("batch/job")),
+			},
+			filter:                client.MatchingFields{GetOwnerKey(kubeflow.SchemeGroupVersionKind): "alpha"},
+			wantFieldMatcherError: true,
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			ctx := context.Background()
+			builder := utiltesting.NewClientBuilder()
+			builder.WithObjects(append(tc.wls, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: testNamespace}})...)
+
+			gotIndexerErr := SetupIndexes(ctx, utiltesting.AsIndexer(builder), tc.opts...)
+			if diff := cmp.Diff(tc.wantError, gotIndexerErr, cmpopts.EquateErrors()); len(diff) != 0 {
+				t.Fatalf("Unexpected setupIndexer error (-want,+got):\n%s", diff)
+			}
+
+			k8sClient := builder.Build()
+
+			gotWls := &kueue.WorkloadList{}
+			if tc.wantFieldMatcherError {
+				// Given that the `wantFieldMatcherError` is `true`, a list operation without fieldMatcher should succeed.
+				if gotListErr := k8sClient.List(ctx, gotWls, client.InNamespace(testNamespace)); gotListErr != nil {
```

Comment:
> IIUC, this operation would succeed also when wantFieldMatcherError=false, right?

Yes, that's right.

> the purpose of this check

I wanted to verify that this operation would succeeded when we don't use the `fieldMatcher`.
We probably also should verify that the `gotWls` has the proper workloads here.

> making it conditional?

I wanted to avoid unnecessary checks when the `wantFieldMatcherError=false`.

Reply:
> We probably also should verify that the gotWls has the proper workloads here.

I think so.

> I wanted to avoid unnecessary checks when the wantFieldMatcherError=false.

I see, but maybe we can remove the conditional for simplicity. Or, alternatively, add a comment why it is checked.

Reply:
updated.